Server crash when Nether Portal was lit with a dispenser + flint & steel while an explosion occurred nearby

```
net.minecraft.server.v1_15_R1.ReportedException: Exception while ticking
        at net.minecraft.server.v1_15_R1.MinecraftServer.b(MinecraftServer.java:1258) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.DedicatedServer.b(DedicatedServer.java:430) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.MinecraftServer.a(MinecraftServer.java:1112) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:934) ~[patched_1.15.2.jar:git-Paper-151]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_242]
Caused by: java.lang.IllegalArgumentException: Cannot get property BlockStateDirection{name=facing, clazz=class net.minecraft.server.v1_15_R1.EnumDirection, values=[north, east, south, west, up, down]} as it does not exist in Block{minecraft:air}
        at net.minecraft.server.v1_15_R1.BlockDataAbstract.get(BlockDataAbstract.java:91) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.DispenseBehaviorItem.dispense(DispenseBehaviorItem.java:18) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.BlockDispenser.dispense(BlockDispenser.java:63) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.BlockDispenser.tick(BlockDispenser.java:89) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.IBlockData.a(IBlockData.java:210) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.WorldServer.b(WorldServer.java:738) ~[patched_1.15.2.jar:git-Paper-151]
        at com.destroystokyo.paper.server.ticklist.PaperTickList.tick(PaperTickList.java:296) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.TickListServer.b(TickListServer.java:46) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.WorldServer.doTick(WorldServer.java:418) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.MinecraftServer.b(MinecraftServer.java:1245) ~[patched_1.15.2.jar:git-Paper-151]
        ... 4 more
[03:17:53 ERROR]:       Cause of unexpected exception was
java.lang.IllegalArgumentException: Cannot get property BlockStateDirection{name=facing, clazz=class net.minecraft.server.v1_15_R1.EnumDirection, values=[north, east, south, west, up, down]} as it does not exist in Block{minecraft:air}
        at net.minecraft.server.v1_15_R1.BlockDataAbstract.get(BlockDataAbstract.java:91) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.DispenseBehaviorItem.dispense(DispenseBehaviorItem.java:18) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.BlockDispenser.dispense(BlockDispenser.java:63) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.BlockDispenser.tick(BlockDispenser.java:89) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.IBlockData.a(IBlockData.java:210) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.WorldServer.b(WorldServer.java:738) ~[patched_1.15.2.jar:git-Paper-151]
        at com.destroystokyo.paper.server.ticklist.PaperTickList.tick(PaperTickList.java:296) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.TickListServer.b(TickListServer.java:46) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.WorldServer.doTick(WorldServer.java:418) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.MinecraftServer.b(MinecraftServer.java:1245) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.DedicatedServer.b(DedicatedServer.java:430) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.MinecraftServer.a(MinecraftServer.java:1112) ~[patched_1.15.2.jar:git-Paper-151]
        at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:934) ~[patched_1.15.2.jar:git-Paper-151]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_242]
```

### Actions to reproduce (if known):
Make a dispenser light a Nether Portal, and spawn an explosion at the same time at the same location. Seems to only occur when explosion is timed to remove blocks at the same tick (or close to same tick) as the portal/redstone contraption lighting the portal.

### Paper build number:
Paper 151

--- a/Spigot-Server-Patches/0558-Don-t-require-FACING-data.patch
+++ b/Spigot-Server-Patches/0558-Don-t-require-FACING-data.patch
@@ -1,0 +1,35 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Mariell Hoversholm <proximyst@proximyst.com>
+Date: Sun, 23 Aug 2020 19:01:04 +0200
+Subject: [PATCH] Don't require FACING data
+
+
+diff --git a/src/main/java/net/minecraft/server/DispenseBehaviorItem.java b/src/main/java/net/minecraft/server/DispenseBehaviorItem.java
+index ca5eae350b971e93df708a29c426609423645430..40ac372281ba19cb776c14ffc957481367148f27 100644
+--- a/src/main/java/net/minecraft/server/DispenseBehaviorItem.java
++++ b/src/main/java/net/minecraft/server/DispenseBehaviorItem.java
+@@ -7,20 +7,22 @@ import org.bukkit.event.block.BlockDispenseEvent;
+ // CraftBukkit end
+ 
+ public class DispenseBehaviorItem implements IDispenseBehavior {
++    private EnumDirection enumdirection; // Paper
+ 
+     public DispenseBehaviorItem() {}
+ 
+     @Override
+     public final ItemStack dispense(ISourceBlock isourceblock, ItemStack itemstack) {
++        enumdirection = isourceblock.getBlockData().get(BlockDispenser.FACING); // Paper - cache facing direction
+         ItemStack itemstack1 = this.a(isourceblock, itemstack);
+ 
+         this.a(isourceblock);
+-        this.a(isourceblock, (EnumDirection) isourceblock.getBlockData().get(BlockDispenser.FACING));
++        this.a(isourceblock, enumdirection); // Paper - cache facing direction
+         return itemstack1;
+     }
+ 
+     protected ItemStack a(ISourceBlock isourceblock, ItemStack itemstack) {
+-        EnumDirection enumdirection = (EnumDirection) isourceblock.getBlockData().get(BlockDispenser.FACING);
++        // Paper - cached enum direction
+         IPosition iposition = BlockDispenser.a(isourceblock);
+         ItemStack itemstack1 = itemstack.cloneAndSubtract(1);
+ 

